src/main/resources/static/templates/SSRFVulnerability/LEVEL_1/SSRF.js

```diff
@@ -1,39 +1,40 @@
-getData();
-
 function setDataInProjectsResponseDiv(data) {
+  let projectsResponseDiv = document.getElementById("projectsResponse");
+  projectsResponseDiv.innerHTML = ''; // Clear any existing content
+
   if (data.isValid) {
     let projectNameAndUrls = JSON.parse(data.content);
-    let tableInformation = '<table id="InfoTable">';
+    let table = document.createElement('table');
+    table.id = "InfoTable";
+
     if (projectNameAndUrls.length > 0) {
+      let headerRow = table.insertRow(-1);
       for (let key in projectNameAndUrls[0]) {
-        tableInformation =
-          tableInformation + '<th class="InfoColumn">' + key + "</th>";
+        let headerCell = document.createElement('th');
+        headerCell.className = "InfoColumn";
+        headerCell.textContent = key; // Safely add text content
+        headerRow.appendChild(headerCell);
       }
     }
 
     projectNameAndUrls.forEach((projNameAndUrl) => {
-      tableInformation =
-        tableInformation +
-        "<tr><td class='InfoColumn'>" +
-        projNameAndUrl.name +
-        "</td><td class='InfoColumn'>" +
-        projNameAndUrl.url +
-        "</td></tr>";
+      let row = table.insertRow(-1);
+      let nameCell = row.insertCell(-1);
+      nameCell.className = 'InfoColumn';
+      nameCell.textContent = projNameAndUrl.name; // Safely add text content
+
+      let urlCell = row.insertCell(-1);
+      urlCell.className = 'InfoColumn';
+      // If you need to create a link, do it safely like this:
+      let link = document.createElement('a');
+      link.textContent = projNameAndUrl.url;
+      link.href = projNameAndUrl.url;
+      // Ensure that any URLs are properly sanitized and validated before adding them to the DOM
+      urlCell.appendChild(link);
     });
-    tableInformation = tableInformation + "</tr>";
-    document.getElementById("projectsResponse").innerHTML = tableInformation;
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities